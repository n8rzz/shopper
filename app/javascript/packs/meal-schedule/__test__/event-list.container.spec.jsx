import React from 'react';
import moment from 'moment';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { EventListContainer } from '../event-list.container';
import {
    assemblyMapMock,
    currentDateMock,
    itemMapMock,
    mealScheduleListMock,
} from '../../__mocks__/mocks';

const currentDateFixture = moment(currentDateMock);

describe('EventListContainer', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<EventListContainer
            assemblies={assemblyMapMock}
            currentDate={currentDateFixture}
            items={itemMapMock}
            mealSchedules={mealScheduleListMock}
            onClickRemoveHandler={sinon.spy()}
        />);

        expect(component).toMatchSnapshot();
    });
});
