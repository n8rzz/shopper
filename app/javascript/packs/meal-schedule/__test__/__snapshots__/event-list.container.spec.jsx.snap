// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListContainer renders correctly with valid props 1`] = `
<div
  className="eventList"
>
  <div
    className="stereo"
  >
    <h3
      className="eventList-hd"
    >
      September
    </h3>
    <div>
      <a
        className="link"
        href="meal_schedules/new"
      >
        New Schedule
      </a>
    </div>
  </div>
  <ul
    className="eventList-bd"
  >
    <DayScheduleList
      assemblies={
        Object {
          "1": Object {
            "created_at": "2019-07-29T22:49:47.431Z",
            "id": 1,
            "name": "Souvlaki",
            "updated_at": "2019-07-29T22:49:47.431Z",
          },
          "2": Object {
            "created_at": "2019-07-29T22:49:47.519Z",
            "id": 2,
            "name": "Chilli con Carne",
            "updated_at": "2019-07-29T22:49:47.519Z",
          },
        }
      }
      daySchedule="2019-09-24T00:00:00.000Z"
      items={
        Object {
          "13": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 3,
            "id": 13,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
          "15": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 5,
            "id": 15,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
        }
      }
      key="daysSchedule-0"
      mealScheduleMap={
        Object {
          "2019-09-24T00:00:00.000Z": Array [
            Object {
              "assembly_id": null,
              "created_at": "2019-09-04T01:00:35.645Z",
              "id": 1,
              "item_id": 15,
              "meal_time": "none",
              "order_item_id": null,
              "purchased": false,
              "schedule_date": "2019-09-24T00:00:00.000Z",
              "updated_at": "2019-09-04T01:00:35.645Z",
            },
            Object {
              "assembly_id": 2,
              "created_at": "2019-09-04T01:00:35.645Z",
              "id": 2,
              "item_id": null,
              "meal_time": "none",
              "order_item_id": null,
              "purchased": false,
              "schedule_date": "2019-09-24T00:00:00.000Z",
              "updated_at": "2019-09-04T01:00:35.645Z",
            },
          ],
          "2019-09-29T00:00:00.000Z": Array [
            Object {
              "assembly_id": 2,
              "created_at": "2019-09-03T01:00:35.645Z",
              "id": 3,
              "item_id": null,
              "meal_time": "none",
              "order_item_id": null,
              "purchased": false,
              "schedule_date": "2019-09-29T00:00:00.000Z",
              "updated_at": "2019-09-07T01:00:35.645Z",
            },
          ],
        }
      }
      onClickRemoveHandler={[Function]}
    />
    <DayScheduleList
      assemblies={
        Object {
          "1": Object {
            "created_at": "2019-07-29T22:49:47.431Z",
            "id": 1,
            "name": "Souvlaki",
            "updated_at": "2019-07-29T22:49:47.431Z",
          },
          "2": Object {
            "created_at": "2019-07-29T22:49:47.519Z",
            "id": 2,
            "name": "Chilli con Carne",
            "updated_at": "2019-07-29T22:49:47.519Z",
          },
        }
      }
      daySchedule="2019-09-29T00:00:00.000Z"
      items={
        Object {
          "13": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 3,
            "id": 13,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
          "15": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 5,
            "id": 15,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
        }
      }
      key="daysSchedule-1"
      mealScheduleMap={
        Object {
          "2019-09-24T00:00:00.000Z": Array [
            Object {
              "assembly_id": null,
              "created_at": "2019-09-04T01:00:35.645Z",
              "id": 1,
              "item_id": 15,
              "meal_time": "none",
              "order_item_id": null,
              "purchased": false,
              "schedule_date": "2019-09-24T00:00:00.000Z",
              "updated_at": "2019-09-04T01:00:35.645Z",
            },
            Object {
              "assembly_id": 2,
              "created_at": "2019-09-04T01:00:35.645Z",
              "id": 2,
              "item_id": null,
              "meal_time": "none",
              "order_item_id": null,
              "purchased": false,
              "schedule_date": "2019-09-24T00:00:00.000Z",
              "updated_at": "2019-09-04T01:00:35.645Z",
            },
          ],
          "2019-09-29T00:00:00.000Z": Array [
            Object {
              "assembly_id": 2,
              "created_at": "2019-09-03T01:00:35.645Z",
              "id": 3,
              "item_id": null,
              "meal_time": "none",
              "order_item_id": null,
              "purchased": false,
              "schedule_date": "2019-09-29T00:00:00.000Z",
              "updated_at": "2019-09-07T01:00:35.645Z",
            },
          ],
        }
      }
      onClickRemoveHandler={[Function]}
    />
  </ul>
</div>
`;
